lib/ts/recipe/session/sessionClass.ts

```diff
@@ -162,4 +139,93 @@ export default class Session implements SessionContainerInterface {
             throw err;
         }
     };
+
+    getSessionClaimPayload() {
+        return this.claims;
+    }
+
+    async updateClaim(claim: SessionClaim<any>, userContext?: any): Promise<void> {
+        await this.updateClaims([claim], userContext);
+    }
+
+    async updateClaims(claims: SessionClaim<any>[], userContext?: any): Promise<void> {
+        const origSessionClaimPayloadJSON = JSON.stringify(this.getSessionClaimPayload());
+
+        let newSessionClaimPayload = this.getSessionClaimPayload();
+        for (const claim of claims) {
+            const value = await claim.fetch(this.getUserId(), userContext);
+            if (value !== undefined) {
+                newSessionClaimPayload = claim.addToPayload(this.getSessionClaimPayload(), value, userContext);
+            }
+        }
+
+        if (JSON.stringify(newSessionClaimPayload) !== origSessionClaimPayloadJSON) {
+            await this.regenerateToken(undefined, newSessionClaimPayload, userContext);
+        }
+    }
+    checkClaimInToken(claim: SessionClaim<any>, userContext?: any): Awaitable<boolean> {
+        return claim.isValid(this.getSessionClaimPayload(), userContext);
+    }
+    async addClaim<T>(claim: SessionClaim<T>, value: T, userContext?: any): Promise<void> {
+        const newSessionClaimPayload = claim.addToPayload(this.getSessionClaimPayload(), value, userContext);
+
+        let newAccessTokenPayload;
+        if (this.helpers.config.jwt.enable && claim.updateAccessTokenPayload) {
+            newAccessTokenPayload = claim.updateAccessTokenPayload(this.getAccessTokenPayload(), value, userContext);
+        }
+        await this.regenerateToken(newAccessTokenPayload, newSessionClaimPayload, userContext);
+    }
+    async removeClaim<T>(claim: SessionClaim<T>, userContext?: any): Promise<void> {
+        const newSessionClaimPayload = claim.removeFromPayload(this.getSessionClaimPayload(), userContext);
+
+        let newAccessTokenPayload;
+        if (this.helpers.config.jwt.enable && claim.updateAccessTokenPayload) {
+            newAccessTokenPayload = claim.updateAccessTokenPayload(
+                this.getAccessTokenPayload(),
+                undefined,
+                userContext
+            );
+        }
```

Comment:
This is being done in the with-jwt sessionClass already (and that doesn't call originalImplementation.removeClaim - so this condition will never be true?